ee/tables/execparsers/data_table/parser.go

```diff
@@ -0,0 +1,92 @@
+package data_table
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+// data_table is a general parser for an input of data which conforms to columns and rows styled output.
+// Parser options
+// skipLines - The number of initial lines of data to skip. By default no lines are skipped. This can be useful if consistent undesired output/garbage is printed before the data to parse.
+// headers - The set of headers. If left blank, the parser assumes the headers are in the first line of data and splits the line to set them.
+// delimiter - The splitting string. If left blank, the parser assumes the delimiter is whitespace and uses `strings.Fields()` split method.
+type parser struct {
+	skipLines uint
```

Comment:
What case uses skipLines? I can imagine some that want it, like that `top` test data. But I wonder if you'd see more reliability by defining either a comment character (some outputs use `#` for comment style lines), or a stanza deliminator (in that top example, there's an `\n\n` between the header and the content.

Reply:
I didn't have a lot of use cases in mind, but I was actively aware that in some instances I would want to skip initial lines. For example, I could switch the [apt](https://github.com/kolide/launcher/blob/main/ee/tables/execparsers/apt/test-data/apt_upgradeable.txt) or [dnf](https://github.com/kolide/launcher/blob/main/ee/tables/execparsers/dnf/test-data/dnf_upgradeable.txt) upgradeable parsers to use this one, but I would want to skip the initial lines they both output.

I agree that either the comment character or stanza delimiter could be a better pattern for this going forward.